### PR TITLE
Add functions to create mutable dict/array from JSON. CBL-2358

### PR DESCRIPTION
**Pull request by jianminzhao**
The new mutables are resursively mutable, that is, all nested arrays and dicts are also mutable.

### CODE REVIEW

File: API/fleece/Fleece.h [hunk 1]
```diff
@@ -484,6 +484,10 @@ while (NULL != (value = FLArrayIterator_GetValue(&iter))) {
         Its initial ref-count is 1, so a call to FLMutableArray_Free will free it.  */
     FLMutableArray FLMutableArray_New(void) FLAPI;
 
+    /** Creates a new mutable Array from JSON. The input json must represent a JSON array or NULL will be returned.
+        Its initial ref-count is 1, so a call to FLMutableArray_Free will free it.  */
```
Comment (pasin) [inline API/fleece/Fleece.h:488]: FLMutableArray_Release instead of FLMutableArray_Free?

File: API/fleece/Fleece.h [hunk 2]
```diff
@@ -718,6 +722,10 @@ while (NULL != (value = FLDictIterator_GetValue(&iter))) {
         Its initial ref-count is 1, so a call to FLMutableDict_Free will free it.  */
     FLMutableDict FLMutableDict_New(void) FLAPI;
 
+    /** Creates a new mutable Dict from json. The input JSON must represent a JSON array, or NULL will be returned.
+        Its initial ref-count is 1, so a call to FLMutableDict_Free will free it.  */
```
Comment (pasin) [inline API/fleece/Fleece.h:726]: FLMutableDict_Release instead of FLMutableDict_Free?

**Review by pasin — APPROVED** [review]
Have small comment about the API doc but the rest looks good to me.

**Review by pasin — COMMENTED** [review]
Can you fix the build error?

File: API/fleece/Fleece.h (hunk 2 above)
Comment (jianminzhao) [reply API/fleece/Fleece.h]: yes. copy/paste error. Fixed

File: API/fleece/Fleece.h (hunk 1 above)
Comment (jianminzhao) [reply API/fleece/Fleece.h]: fixed

File: Fleece/API_Impl/Fleece.cc [hunk 3]
```diff
@@ -229,6 +228,31 @@ FLMutableArray FLMutableArray_New(void) FLAPI {
     return _newMutableArray(nullptr, kFLDefaultCopy);
 }
 
+FLMutableArray FLMutableArray_NewFromJSON(FLString json, FLError* outError) FLAPI {
+    if (outError != nullptr) {
+        *outError = kFLNoError;
+    }
```
Comment (snej) [inline Fleece/API_Impl/Fleece.cc:234]: FYI, these lines aren't necessary -- in our calling conventions, if the function returns a success value, the contents of `*outError` are explicitly undefined. (You don't need to take them out, just letting you know.)

**Review by snej — APPROVED** [review]

**Pull request merged**